archeryutils/constants.py

```diff
@@ -1,6 +1,9 @@
 """Constants used in the archeryutils package."""
 
-from typing import ClassVar
+from collections.abc import Collection, Set
+from typing import ClassVar, TypeVar, Union
+
+T = TypeVar("T")
```

Comment:
I also use the same function to parse the units in `from_face_spec` where its a dict, and at that point it felt like it made more sense to make it truely generic rather than encoding all the known use cases into its type signature.

Reply:
I don't quite follow, I think I didn't word well, sorry.

I was referring to the `TypeVar("T")` - I only see this used in `parse_optional_units()` where it seems to me you could use `float` instead of `T`? Did it get left in from a previous iteration of code?

Reply:
Ah, right. No to be clear I mean that inside `Target.from_face_spec` I call `length.parse_optional_units(face_spec)`, so here the input argument could be a dict instead of a float.

Reply:
Ah, cool, gotcha!